File: 1_two_sum.py
# -*- coding: utf-8 -*-

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return False
        diff_dict = {}
        for i in range(len(nums)):
            if nums[i] in diff_dict:
                return [diff_dict[nums[i]], i]
            else:
                diff_dict[target - nums[i]] = i

s = Solution()
print s.twoSum([2,7,11,15], 17)

File: 1_two_sum.java
package org.cjy.leetcode;

import java.util.*;

class Solution{
	public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer,Integer> map = new HashMap<Integer,Integer>();
        for(int i=0; i<nums.length; i++){
            if(map.containsKey(target - nums[i])){
                result[0] = map.get(target - nums[i]);
                result[1] = i;
                return result;
            }
            map.put(nums[i], i);
        }
        return result;
    }
}

public class SolutionDemo {
	public static void main(String[] args) {
		Solution s = new Solution();
		int[] nums = {2,7,11,15};
		int target = 17;
		int[] result = s.twoSum(nums, target);
		
		for(int i=0; i<result.length; i++){
			System.out.print(result[i] + " ");
		}
	}
}
